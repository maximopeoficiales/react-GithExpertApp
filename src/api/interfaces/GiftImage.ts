export default interface GiftImage {
    id: string,
    title: string,
    bitly_url: string,
    imgUrl: string,
}