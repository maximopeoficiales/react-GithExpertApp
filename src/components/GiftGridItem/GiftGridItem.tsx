import React from "react";
import GiftImage from "../../api/interfaces/GiftImage";
interface MyProps {
  image: GiftImage;
}
const defaultProps = {};
const GiftGridItem = (props: MyProps) => {
  props = { ...defaultProps, ...props };
  const { image } = props;

  return (
    <div data-testid="GiftGridItem" className="card-gift  ">
      <img src={image.imgUrl} alt="" />
      <a href={image.bitly_url}>
        <h1>{image.title}</h1>
      </a>
    </div>
  );
};

GiftGridItem.propTypes = {};

export default GiftGridItem;
