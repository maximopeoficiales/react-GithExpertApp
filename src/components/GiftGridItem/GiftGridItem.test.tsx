import React from "react";
import "@testing-library/jest-dom/extend-expect";
import GiftGridItem from "./GiftGridItem";
import { shallow } from "enzyme";
import GiftImage from "../../api/interfaces/GiftImage";

describe("<GiftGridItem />", () => {
  const image: GiftImage = {
    id: "2",
    bitly_url:
      "https://i.picsum.photos/id/189/200/200.jpg?hmac=D4dFU1JWalnD5ZptSD-jGDOEc8abV1vBRfC5HstzT_8",
    title: "algun titulo",
    imgUrl: "imge...",
  };
  const wrapper = shallow(<GiftGridItem image={image} />);
  test("debe de mostrar <GiftGridItem/>", () => {
    // verifica si se renderizo correctamente
    expect(wrapper).toMatchSnapshot();
  });
  test("debe mostrar los datos identicos a los enviados por props", () => {
    const img = wrapper.find("img");
    const enlace = wrapper.find("a");
    const h1 = wrapper.find("h1");
    // console.log(img.props());
    // prop() busca en las propiedades del elemento
    expect(img.prop("src")).toBe(image.imgUrl);
    expect(enlace.prop("href")).toBe(image.bitly_url);
    expect(h1.text().trim()).toBe(image.title);
  });
  test("debe tener el div la clase card-gift", () => {
    const div = wrapper.find("div");
    const classes = div.prop("className");
    expect(classes?.includes("card-gift")).toBe(true);
  });
});
