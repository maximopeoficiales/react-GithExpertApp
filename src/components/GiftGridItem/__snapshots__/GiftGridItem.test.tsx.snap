// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GiftGridItem /> debe de mostrar <GiftGridItem/> 1`] = `
<div
  className="card-gift  "
  data-testid="GiftGridItem"
>
  <img
    alt=""
    src="imge..."
  />
  <a
    href="https://i.picsum.photos/id/189/200/200.jpg?hmac=D4dFU1JWalnD5ZptSD-jGDOEc8abV1vBRfC5HstzT_8"
  >
    <h1>
      algun titulo
    </h1>
  </a>
</div>
`;
